ppafm/core.py

```diff
@@ -150,14 +150,28 @@ def setTip(lRadial=None, kRadial=None, rPP0=None, kSpring=None, parameters=None)
     lib.setTip(lRadial, kRadial, rPP0, kSpring)
 
 
-# void setTipSpline( int n, double * xs, double * ydys ){
-lib.setTipSpline.argtypes = [c_int, array1d, array2d]
-lib.setTipSpline.restype = None
+# struct SplineParameters {
+#     int      rff_n;
+#     double * rff_xs;
+#     double * rff_ydys;
+# }
+class SplineParameters(Structure):
+    _fields_ = [
+        ("rff_n", c_int),
+        ("rff_xs", POINTER(c_double)),
+        ("rff_ydys", POINTER(c_double)),
+    ]
 
+    def __init__(self, xs, ydys):
+        super().__init__()
+        self.rff_n = len(xs)
+        self.rff_xs = np.ctypeslib.as_ctypes(xs.astype(np.double))
```

Comment:
I originally tried to do it with the `np.ctypeslib.ndpointer` so that you could just use the numpy array, but for some reason that does not work with structs, so I opted to use the `ctypes` pointer instead which works.

I changed it so that the original numpy arrays are stored there as well in a separate variables starting with `np_`.